### PR TITLE
Improved RDS S3 script to use shared security group for RDS db instan…

### LINKED ISSUE #137: Avoid  Ec2 VPC SecurityGroups Quota When Setting Up RDS
**Is your feature request related to a problem? Please describe.**
When trying to setup Kubeflow using `main` + `v1.4.1` upstream, I ran into an issue at the db creation step.  
`boto3` returned the following error : `An error occurred when calling the CreateDBInstance operation: The request violates your limit on number of Ec2 VPC SecurityGroups associated with an instance`   
After hours of back and forth with the support, no solution was found (we increased a lot of quotas related to EC2, VPCs etc but nothing worked).
  
After some more debugging I found the following quota : 
The default quota for the number of VPC Security Groups that can be attached to an RDS instance is 5 by default and this quota is not adjustable :    
securitygroupquota
I verified the number of VPC Security Group that was associated to the db instance during the db instance creation step and ours was 6, so just 1 over the limit.

**Describe the solution you'd like**  
To ensure all nodes have access to the database, instead of associating all the security groups to the database, we can associate just 1 security group called `ClusterSharedNodeSecurityGroup` that is meant to share communication with all nodes.  
This will ensure that no quota limit is reached regardless of your node count.

### PR DESCRIPTION
**Pull request by AlexandreBrown**
**Which issue is resolved by this Pull Request:**
Resolves #137  

- [x] End-to-end Test on fresh cluster using `main` + `v1.4.1` and `cognito-rds-s3` setup
- [x] Verify that nodes have access to the RDS DB (KFP can push artifacts, db is populated etc)

### CODE REVIEW

File: docs/deployment/rds-s3/auto-rds-s3-setup.py [hunk 1]
```diff
@@ -372,7 +372,7 @@ def get_vpc_id(vpc):
     return list(map(get_vpc_id, vpcs))
```
Comment (mbaijal) [inline docs/deployment/rds-s3/auto-rds-s3-setup.py:374]: Could you add a comment here explaining the context

**Review by mbaijal — APPROVED** [review]
Could you add comment to the script explaining why we are picking only one SG. Otherwise looks ok.

File: docs/deployment/rds-s3/auto-rds-s3-setup.py (hunk 1 above)
Comment (AlexandreBrown) [reply docs/deployment/rds-s3/auto-rds-s3-setup.py:374]: Done!

**Review by rrrkharse — APPROVED** [review]

**Comment by surajkota** [comment]
We need to validate these changes with tests. @AlexandreBrown could you help us move this script to `tests/e2e/utils` similar to #106. Feel free to change the test code as well to use these script

we are working on PR builds which should help with automated tests

**Pull request merged**